File: Data Access/EF-CodeFirst/CodeFirst/Department.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeFirst
{
    [Table("Department", Schema = "HR")]
    class Department
    {
        public int ID { get; set; }
        [Required, MaxLength(100)]
        public string Name { get; set; }
        [Required, MaxLength(100)]
        public string Location { get; set; }

        [InverseProperty("Department")]

        public virtual ICollection<Employee> Employees { get; set; }

        [InverseProperty("SupervisedDept")]
        public virtual ICollection<Employee> Supervisors { get; set; }

        public virtual ICollection<Project> Projects { get; set; }
    }
}

File: Angular/Labs/Lab4/src/app/_models/department.ts
// export class Department {
//     id : number;
//     deptName : string;
//     location : string;
//     constructor(id:number,deptName:string,location:string)  {
//       this.id = id;
//       this.deptName = deptName;
//       this.location = location;
//     }
// }
export class Department {
  constructor(public id:number = 0,public deptName:string = '',public location:string = '')  {

  }
}
